src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -434,9 +434,9 @@ public static partial class MarshalHelpers
                     return MarshallerKind.Invalid;
                 }
 
-                if (!isField && InteropTypes.IsInt128Type(context, type))
+                if (!isField && ((DefType)type).IsInt128OrHasInt128Fields)
```

Comment:
I'll defer to you here, but how are we sure this will always be a `DefType`?

Reply:
It's in a IsValueType branch. Value types are always `DefType`.